code/src/layouts/header/components/header-toolbar/HeaderToolbar.tsx

```diff
@@ -32,10 +35,13 @@ const HeaderToolbar = () => {
   const { openModal } = useModalContext();
   const isAuthenticated = useIsAuthSelector();
   const isLoadingAuth = useIsAuthLoadingSelector();
+  const userDetails = useUserDetailsSelector();
   const dispatch = useAppDispatch();
 
   const [searchValue, setSearchValue] = useState("");
 
+  const { role } = userDetails || {};
```

Comment:
It is not the best solution here, because role can be null and we will be forces write this code over and over again. Makes sense to create useUserRoleSelector, that can look like following:

```jsx
export const useUserRoleSelector = () =>
  useAppSelector((store) => {
    const userDetails = store.user.userDetails;

    if (userDetails === null) {
      return null;
    }

    return userDetails.role;
  });
```

role can be `null` if user is not authorized, so we should take it into account

Reply:
Fixed like you suggests